parser/src/powdr.lalrpop

```diff
@@ -744,21 +754,22 @@ TraitFunction: TraitFunction<Expression> = {
 }
 
 TraitImplementation: TraitImplementation<Expression> = {
-    "impl" <type_scheme: GenericTraitName> "{" <functions:NamedExpressions> "}" => TraitImplementation { name: type_scheme.0, type_scheme: type_scheme.1, functions }
+    <start:@L> "impl" <type_scheme: GenericTraitName> "{" <functions:NamedArcExpressions> "}" <end:@L> => TraitImplementation { name: type_scheme.0, source_ref: ctx.source_ref(start, end), type_scheme: type_scheme.1, functions }
 }
 
-NamedExpressions: Vec<NamedExpression<Expression>> = {
+
+NamedArcExpressions: Vec<NamedExpression<Arc<Expression>>> = {
     => vec![],
-    <mut list:( <NamedExpression> "," )*> <end:NamedExpression> ","?  => { list.push(end); list }
+    <mut list:( <NamedArcExpression> "," )*> <end:NamedArcExpression> ","?  => { list.push(end); list }
 }
 
-NamedExpression: NamedExpression<Expression> = {
-    <name:Identifier> ":" <body:BoxedExpression> => NamedExpression { name, body }
+NamedArcExpression: NamedExpression<Arc<Expression>> = {
```

Comment:
I had to duplicate the `NamedExpressions` because they are also used in the structs, where Arc is not necessary (it uses Box). I don't think it's worth the cost to unify them.